addons/controllers/clusterbootstrap_controller.go

```diff
@@ -1174,6 +1139,8 @@ func (r *ClusterBootstrapReconciler) GetDataValueSecretNameFromBootstrapPackage(
 				return packageSecretName, nil
 			}
 		}
+		// cbPkg.ValuesFrom is nil and not TKGS
+		return "", nil
 	}
 
 	// When valuesFrom is not nil, but either valuesFrom.Inline, valuesFrom.SecretRef, or valuesFrom.providerRef is empty or nil,
```

Comment:
@vijaykatam , I think I should have removed this part and just return nil here. This section seems unreachable to me

Reply:
We need a catch all to return an error otherwise the func won't compile.